R/MsBackendMzR.R

```diff
@@ -49,17 +52,24 @@ setMethod("readMsObject", signature(object = "MsBackendMzR",
               fl <- file.path(param@path, "backend_data.txt")
               if (!file.exists(fl))
                   stop("No 'backend_data.txt' file found in the provided path.")
-              data <- read.table(file = fl, sep = "\t", header = TRUE)
-              rownames(data) <- NULL
-              data <- DataFrame(data)
-              object@spectraData <- data
-              if (length(spectraPath) > 0) {
-                  old <- common_path(dataStorage(object))
-                  dataStoragePaths <- dataStorage(object)
-                  normalizedDataStoragePaths <- normalizePath(dataStoragePaths,
-                                                              winslash = "/")
-                  dataStorage(object) <- sub(old, spectraPath,
-                                        normalizedDataStoragePaths)
+              l2 <- readLines(fl, n = 2)
+              if (l2[1] != "# MsBackendMzR")
+                  stop("Invalid class in 'backend_data.txt' file.",
+                  "Should run with object = ", l2[1])
+              if (!is.na(l2[2])) { # or length(l2) > 1 ?
+                  data <- read.table(file = fl, sep = "\t", header = TRUE)
+                  rownames(data) <- NULL
+                  data <- DataFrame(data)
+                  object@spectraData <- data
```

Comment:
Maybe combine here the two lines and use directly `object@spectraData <- DataFrame(data)`? then R would not need to first replace the `data` variable with a new object (`DataFrame`) before assigning it?